### PR TITLE
add specialUse reserved

### PR DESCRIPTION
**Pull request by loxp**
UCP #2176

### What problem does this PR solve? 
UCP #2176

### What is changed and how it works?
add specialUse label 'reserved'

### Check List 

Tests 

 - Unit test

Code changes

 - Has configuration change

Side effects

Related changes

 - Need to update the documentation

### CODE REVIEW

File: server/schedule/filter/filters.go [hunk 1]
```diff
@@ -583,6 +583,7 @@ func (f *specialUseFilter) Target(opt opt.Options, store *core.StoreInfo) bool {
 const (
 	specialUseKey       = "specialUse"
 	specialUseHotRegion = "hotRegion"
+	specialUseReserved  = "reserved"
 )
 
-var allSpecialUses = []string{specialUseHotRegion}
+var allSpecialUses = []string{specialUseHotRegion, specialUseReserved}
```
Comment (nolouch) [inline server/schedule/filter/filters.go]: Can  `hot region` scheduler balance to the reserved store?

Comment (loxp) [reply server/schedule/filter/filters.go]: SpecialUseFilter is added in `balanceSolver`, then `hot region` scheduler can only balance to stores for hot region.

**Comment by nolouch** [comment]
PTAL @disksing

**Comment by disksing** [comment]
@loxp Thanks for your contribution!

**Pull request merged**